stl/inc/chrono

```diff
@@ -5171,6 +5176,180 @@ namespace chrono {
+        int _Dynamic_precision_index = -1;
+        _Align _Alignment            = _Align::_None;
+        // At most one codepoint (so one char32_t or four utf-8 char8_t)
+        _CharT _Fill[4] = {' ', _CharT{0}, _CharT{0}, _CharT{0}};
+        // recursive definition in grammar, so could have any number of these with literal chars
+        vector<_Chrono_specs<_CharT>> _Chrono_specs_list;
+    };
+
+    // Model of _Chrono_parse_spec_callbacks that fills a _Chrono_format_specs with the parsed data
+    template <class _CharT>
+    class _Chrono_specs_setter {
+    public:
+        constexpr explicit _Chrono_specs_setter(_Chrono_format_specs<_CharT>& _Specs_) : _Specs(_Specs_) {}
+
+        // same as _Specs_setter
+        constexpr void _On_align(_Align _Aln) {
+            _Specs._Alignment = _Aln;
+        }
+
+        // same as _Specs_setter
+        constexpr void _On_fill(basic_string_view<_CharT> _Sv) {
+            if (_Sv.size() > 4) {
+                _THROW(format_error("Invalid fill (too long)."));
+            }
+
+            _STD fill(_Specs._Fill, _Specs._Fill + 4, _CharT{});
+            _STD copy(_Sv.begin(), _Sv.end(), _Specs._Fill);
+        }
+
+        constexpr void _On_width(int _Width) {
+            _Specs._Width = _Width;
+        }
+
+        constexpr void _On_precision(int _Prec) {
+            _Specs._Precision = _Prec;
+        }
+
+        constexpr void _On_conversion_spec(_CharT _Modifier, _CharT _Type) {
+            // NOTE: same performance note from _Basic_format_specs also applies here
+            const char _Char_mod  = static_cast<char>(_Modifier);
+            const char _Char_type = static_cast<char>(_Type);
+            if (_Char_mod != '\0' && _Char_mod != 'E' && _Char_mod != 'O') {
+                _THROW(format_error("Invalid modifier specification."));
+            }
+
+            if (_Type < 0 || _Type > (numeric_limits<signed char>::max)()) {
+                _THROW(format_error("Invalid type specification."));
+            }
+
+            _Chrono_specs<_CharT> _Conv_spec{._Modifier = _Char_mod, ._Type = _Char_type};
+            _Specs._Chrono_specs_list.push_back(_Conv_spec);
+        }
+
+        constexpr void _On_lit_char(_CharT _Lit_ch) {
+            _Chrono_specs<_CharT> _Lit_char_spec{._Lit_char = _Lit_ch};
+            _Specs._Chrono_specs_list.push_back(_Lit_char_spec);
+        }
+
+    protected:
+        _Chrono_format_specs<_CharT>& _Specs;
+    };
+
+    template <class _CharT, _Chrono_parse_spec_callbacks<_CharT> _Callbacks_type>
+    _NODISCARD constexpr const _CharT* _Parse_conversion_specs(
+        const _CharT* _Begin, const _CharT* _End, _Callbacks_type&& _Callbacks) {
+        ++_Begin; // move past %
+        if (_Begin == _End || *_Begin == '}') {
+            _THROW(format_error("Invalid format string."));
+        }
+
+        _CharT _Mod = '\0';
+        _CharT _Ch  = *_Begin;
+
+        if (_Ch == 'E' || _Ch == 'O') { // includes modifier

```

Comment:
Nitpick, I would prefer to have some names for `E` and `O`

Reply:
we discussed this a bit and I suspect it's pretty much just busywork to use enums for them, the standard just uses the characters as well.